### PR TITLE
SeveNChaK

### PR DESCRIPTION
**Pull request by SeveNChaK**
Remove line if task was accepted earlier.

Mark task as done `[x]` if your resolve it.

- [x] 1 ArrayDequeSimple
- [x] 2 ArrayDequeSimple + Iterator Remove
- [x] 3 ArrayDequeFull
- [x] 4 LinkedDequeSimple
- [x] 5 LinkedDequeSimple + Iterator Remove
- [x] 6 LinkedDequeFull
- [x] 7 ArrayPriorityQueueSimple
- [x] 8 ArrayPriorityQueueSimple + Iterator Remove
- [x] 9 MergingPeekingIncreasingIterator
- [x] 10 AVLTree
- [x] 11 AVLTreeIterable
- [x] 12 AVLTreeIterable + Iterator Remove
- [ ] 13 RedBlackTree (add)
- [ ] 14 RedBlackTree (add + remove)
- [x] 15 OpenHashTable + Student
- [x] 16 OpenHashTable + Student + Iterator Remove

### CODE REVIEW

**Comment by mikhail-nechaev** [comment]
целевой бранч должен быть `SeveNChaK`, а не master
нужно изменить `base` у PR

File: src/ru/mail/polis/collections/iterator/todo/IntegerIncreasingSequencePeekingIterator.java [hunk 1]
```diff
@@ -74,6 +108,13 @@ public Integer peek() {
      */
     @Override
     public int compareTo(IPeekingIterator<Integer> other) {
-        throw new UnsupportedOperationException("todo: implement this");
+        if (hasNext() && other.hasNext()) {
+            return peek() - other.peek();
```
Comment (mikhail-nechaev) [inline src/ru/mail/polis/collections/iterator/todo/IntegerIncreasingSequencePeekingIterator.java]: это неправильно из-за возможного переполнения

File: src/ru/mail/polis/collections/iterator/todo/MergingPeekingIncreasingIterator.java [hunk 2]
```diff
@@ -10,46 +14,59 @@
...
+     * <p>
      * Complexity = O(n)
      *
      * @param iterators the iterators whose are to be placed into this merging peeking increasing iterator
      */
     public MergingPeekingIncreasingIterator(IntegerIncreasingSequencePeekingIterator... iterators) {
-        //todo: do some stuff with iterators
+        queue = new ArrayPriorityQueueSimple<>();
+        for (IntegerIncreasingSequencePeekingIterator iterator : iterators) {
+            queue.add(iterator);
```
Comment (mikhail-nechaev) [inline src/ru/mail/polis/collections/iterator/todo/MergingPeekingIncreasingIterator.java]: это не `Complexity = O(n)`

File: src/ru/mail/polis/collections/list/todo/ArrayPriorityQueueSimple.java [hunk 3]
```diff
@@ -48,64 +53,126 @@ public ArrayPriorityQueueSimple(Comparator<E> comparator) {
...
      * @throws NullPointerException if the specified collection or comparator is null
      */
     public ArrayPriorityQueueSimple(Collection<E> collection, Comparator<E> comparator) {
+        if (collection == null || comparator == null) {
+            throw new NullPointerException();
+        }
         this.comparator = Objects.requireNonNull(comparator, "comparator");
-        //todo: do some stuff with collection
+        for (E o : collection) {
+            add(o);
```
Comment (mikhail-nechaev) [inline src/ru/mail/polis/collections/list/todo/ArrayPriorityQueueSimple.java]: это не `Complexity = O(n)`

File: src/ru/mail/polis/collections/set/hash/todo/OpenHashTable.java [hunk 4]
```diff
@@ -19,6 +20,17 @@
  */
 public class OpenHashTable<E extends IOpenHashTableEntity> implements IOpenHashTable<E> {
```
Comment (mikhail-nechaev) [inline src/ru/mail/polis/collections/set/hash/todo/OpenHashTable.java]: списано #5

Comment (SeveNChaK) [reply src/ru/mail/polis/collections/set/hash/todo/OpenHashTable.java]: Очень стыдно🙈

**Comment by mikhail-nechaev** [comment]
списано: 5584a5c25ea703a0e00a8a5e6dbdebd0342184f7 > f7eeb9e528b11043f0c3c0446934403aea67ead0 > eff0f99c44c8969f01538a92d3264a3d0264942e
- [ ] 10 AVLTree
- [ ] 11 AVLTreeIterable
- [ ] 12 AVLTreeIterable + Iterator Remove

**Pull request merged**